### ISSUE TITLE
potential inconsistency in getState method

### ISSUE THREAD

**Issue by paulvanbladel** [assafg/osiris#7]
In GetState a filter is applied
```
  getState(context) {
        return this.db.getSnapshot(context).then(snapshot => {
            return this.db
                .getEvents(context, snapshot.seq)
                .then(events => {
                    if (snapshot.seq) {
                        **return events.filter(e => e.seq > snapshot.seq || e.isSnapshot);**
                  ...
```
I think that e.isSnapshot never exists. Shouldn't this be: e.data.isSnapshot.

Also, I'm not sure if the next processing is using correctly the returned data from the first then block (so the block with the filter). But for being sure about that, I should first understand the intention of the filter block.
May I presume the idea is that, in the first then block, you want to return the latest snapshot and all events after that latest snapshot. If no snapshot exists, all events are returned.
Now your code works fine since you create every time a shapshot. 
But my assumption would be that it should work also if you comment out the line responsible for taking the snapshot. (or add a second param in the getState method which makes taking snapshots conditional).

**Comment by assafg** [comment]
When a snapshot is created it an `isSnapshot` attribute is added (with a `true` value).
This filter gets all the event's that have been added after the snapshot was created (including the snapshot) and then does the state aggregation.
For better read performance a snapshot should be created as often as possible to reduce the time it takes to aggregate (reduce) - hence creating a snapshot every time. It should also work with less frequent snapshots but I accept that I may have introduced a bug some time :)

**Comment by assafg** [comment]
https://github.com/assafg/osiris/commit/1966e2762ce1af5e48fa8073a7dbfbd1f9526783

**Comment by paulvanbladel** [comment]
@assafg Thanks a lot.
I still think there is a bug, I'll try to isolate it more closely.
Are you sure the filter is ok? I think it should be:

 if (snapshot.seq) {
                        return events.filter(e => e.seq > snapshot.seq || e.**data**.isSnapshot);
                    }

**Comment by assafg** [comment]
You are correct! it should be within data. posting a fix in a few minutes

**Comment by assafg** [comment]
Published v1.2.0 with fixes to the above. Thanks again @paulvanbladel for the valuable input
